backend/tests/Feature/GetAvgSesiionTest.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+namespace Tests\Feature;
+
+use Tests\TestCase;
+use Illuminate\Foundation\Testing\WithFaker;
+use Illuminate\Foundation\Testing\RefreshDatabase;
+use App\Entities\User;
+use App\Entities\Website;
+use App\Entities\Visitor;
+use App\Entities\Session;
+use Illuminate\Support\Collection;
+use Illuminate\Support\Carbon;
+use App\Repositories\EloquentSessionRepository;
+use Tymon\JWTAuth\Facades\JWTAuth;
+
+class GetAvgSessionTest extends TestCase
+{
+
+    public function test_average_session_duration()
+    {
+        $user = factory(User::class)->create();
+        $website = factory(Website::class)->create();
+        $user->website()->save($website);
+        $token = JWTAuth::fromUser($user);
+        $websiteVisitors = factory(Visitor::class, 10)->create(['website_id' => $website->id]);
+        $start_session = Carbon::now();
+
+
+        $geoPositionId = factory(\App\Entities\GeoPosition::class)->create()->id;
+        $oSId = factory(\App\Entities\Os::class)->create()->id;
```

Comment:
test should not be so complicated, why you cannot create session like this factory(Session::class, 10)->create() ?

Reply:
In this case test can not be passed because SessionFactory requires related entities (Visitor, Page etc.) to be created before seeding sessions table.
I want to check in this test successful api response with real sql query.